src/postgraphile/http/__tests__/createPostGraphileHttpRequestHandler-test.js

```diff
@@ -850,5 +853,23 @@ for (const [name, createServerFromHandler] of Array.from(serverCreators)) {
         additionalGraphQLContextFromRequest.mock.calls[0][1],
       ).toBeInstanceOf(http.ServerResponse)
     })
+
+    if (name === 'koa') {
+      test('compatiblity with compression middleware', async () => {
+        const server = createServer(
+          { graphiql: true },
+          {
+            onPreCreate: app => {
+              app.use(compress({
+                threshold: 0
+              }))
+            }
+          }
+        )
+        await request(server)
+          .get('/graphiql')
+          .expect('Content-Encoding', /gzip/)
```

Comment:
It's specifically the GraphiQL assets that failed to serve - the JS and CSS - because they're served via `send` which has issues with Koa.

Reply:
gotcha.  i'm not sure what your expectations are, but 565cf87 adds a download/`send` assertion _with_ the compress plugin enabled.

i also linked this build into my project will compress and a variety of other middlewares, and it's taking well.

Reply:
Yeah; I'm not sure exactly how it was failing - Koa was hanging or something, so the connection never closed. Checking `sendFile` is called is probably not sufficient to test that; but I'm not sure how best to go about doing it other than spinning up a literal server and testing it end-to-end. I think there's still definitely some quirks going on with Koa (particularly around GraphiQL) but I've not managed to make them reproducible yet.

Reply:
Cool.

Well, koa tests are now enabled, a send file test is now inserted, and I have tested this end-to-end with koa.

I think we're g2g.